programming/editors/vscode.md

```diff
@@ -0,0 +1,193 @@
+you'll need to tell it where to find the library files:
+
+1. Open the workspace containing your code.
+2. Open your [workspace settings][workspace-settings] file:
+    * On Windows/Linux - **File** > **Preferences** > **Settings**.
+    * On macOS - **Code** > **Preferences** > **Settings**.
+
+3. Select **Workspace** (rather then **User**).
+4. Search for `python.autoComplete.extraPaths`.
+5. Click "Edit in settings.json".
+6. Add the path to the directory which contains the `sr` directory, within
+   double quotes. The path can be either relative to your project or absolute.
+
+   For example if developing code for the simulator and you have extracted the
+   simulator directory next to your code:
+
+   ```
+   .
+   ├── competition-simulator-<version>
+   │   ├── ...
+   │   ├─ modules
+   │   │   └── sr
+   │   │       └── robot
+   │   │           └── ...
+   │   └─ worlds
+   │       └── Arena.wbt
+   └── robot.py
+   ```
+
+   then the path you should add is `"competition-simulator-<version>/modules/"`.
+
+[workspace-settings]: https://code.visualstudio.com/docs/getstarted/settings#_creating-user-and-workspace-settings
+
+## Interactive Debugging
+
+<div class="info">
+This section refers to a feature which is only available within the simulator.
+<!--
+Pedantic note: yes, you can actually make this work on the kits too (very easily
+it turns out), however the steps to set this up on the kits are a bit different
+so for now we just document the simulator version.
+-->
+</div>
+
+Interactive debugging is a great way to inspect what your code is doing whilst
+the simulator is actually running. It will allow you to inspect the values of
+variables throughout the code and even re-run sections to understand how they
+interact.
+
+There are two initial setup steps (installing `debugpy` and configuring VSCode)
+and then two steps each time you want to debug your code.
+
+### Install `debugpy`
+
+In order for VSCode to be able to debug your code as it runs in the simulator
+you will need to install the [`debugpy`][debugpy] Python package. You will use
+this package in your Python code to signal that it is ready for VSCode to attach
+to it, so it needs to be installed into the same Python environment that Webots
+is configured to use.
+
+``` shell
+python -m pip install debugpy
+```
+
+You may need to use the full path to your `python`, or on some platforms it may
+be called `python3`.
+
+On Windows, if specifying a full path to `python.exe`, you should use the basic
+Command Prompt (and not the terminal within VSCode or PowerShell).
+
+### Configure VSCode
+
+Next, we are going to create a [debug configuration][debug-config] so that
+VSCode knows what you want it to debug.
+
+1. Open the workspace containing your code.
+2. Select the Run view in the sidebar (or press
+   <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd> /
+   <kbd>⌘</kbd> + <kbd>⇧</kbd> + <kbd>D</kbd> on macOS)
+3. Click "create a launch.json file".
```

Comment:
If you haven't opened a folder, the blue text is 'open a folder'  ('create a launch.json file' is much harder to spot.

Reply:
Does the first point on this list not cover this?
> 1. Open the workspace containing your code.

Reply:
d9240f3b904f66a66f1acd008a09f2034f6f59c1 should help emphasise this.